platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/metric/NodeMetrics.java

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.otter.fixtures.metric;
+
+// SPDX-License-Identifier: Apache-2.0
+
+import com.swirlds.metrics.api.Metric;
+import com.swirlds.metrics.api.MetricConfig;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.stream.Collectors;
+import org.hiero.consensus.model.node.NodeId;
+
+/**
+ * In-memory implementation of the {@link Metrics} interface that is tailored for usage in otter
+ * integration tests. It records the full history of every metric value so that assertions can be
+ * executed after a test run.
+ */
+public class NodeMetrics implements Metrics {
```

Comment:
I think this implementation will only be used in tests running on Turtle. I suggest moving this package under the `turtle` package.